src/components/task-quicksearch/PipelineQuickSearch.tsx

```diff
@@ -143,19 +180,46 @@ const Contents: React.FC<
               );
             }
           } else {
-            resolve(
-              savedCallback.current({
-                metadata: { name: item.data.task.name },
-              }),
-            );
-            createArtifactHubTask(
-              selectedVersionUrl,
-              namespace,
-              selectedVersion,
-              isDevConsoleProxyAvailable,
-            ).catch(() =>
-              setFailedTasks([...failedTasks, item.data.task.name]),
-            );
+            // Checking if task with same name already exists, if yes then create with a different name to avoid conflict
```

Comment:
nit: These two blocks are pretty much the same, just using a different model. Lets rather create a callback which will take the model and its parameters